pkg/services/comments/commentmodel/models.go

```diff
@@ -0,0 +1,105 @@
+package commentmodel
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	// ObjectTypeOrg is reserved for future use for per-org comments.
+	ObjectTypeOrg = "org"
+	// ObjectTypeDashboard used for dashboard-wide comments.
+	ObjectTypeDashboard = "dashboard"
+	// ObjectTypeAnnotation used for annotation comments.
+	ObjectTypeAnnotation = "annotation"
+)
+
+var RegisteredObjectTypes = map[string]struct{}{
+	ObjectTypeOrg:        {},
+	ObjectTypeDashboard:  {},
+	ObjectTypeAnnotation: {},
+}
+
+type CommentGroup struct {
+	Id         int64
+	OrgId      int64
+	ObjectType string
+	ObjectId   string
+	Settings   Settings
+
+	Created int64
+	Updated int64
+}
+
+func (i CommentGroup) TableName() string {
+	return "comment_group"
+}
+
+type Settings struct {
+}
+
+var (
+	_ driver.Valuer = Settings{}
+	_ sql.Scanner   = &Settings{}
+)
```

Comment:
have not seen this anywhere else, can you describe what this is?

Reply:
This is a way to work with JSON using a typed approach - i.e. we can use structs in Go, but it will be automatically marshalled to JSON when saving into the database and unmarshalled when retrieving.